### PR TITLE
lualine: add winbar and inactive sections configs

### PR DESCRIPTION
**Pull request by andrevmatos**
Simple PR to expand lualine config to allow winbar and inactive sections and winbar components config.

### CODE REVIEW

File: plugins/statuslines/lualine.nix [hunk 1]
```diff
@@ -79,6 +79,21 @@ with lib; let
...
+          lualine_a = mkComponentOptions "";
+          lualine_b = mkComponentOptions "";
+          lualine_c = mkComponentOptions "";
+
+          lualine_x = mkComponentOptions "";
+          lualine_y = mkComponentOptions "";
+          lualine_z = mkComponentOptions "";
+        };
+      })
+      name;
```
Comment (GaetanLepage) [inline plugins/statuslines/lualine.nix:96]:
```suggestion
    mkEmptySectionOption = name:
      helpers.mkCompositeOption name {
        lualine_a = mkComponentOptions "";
        lualine_b = mkComponentOptions "";
        lualine_c = mkComponentOptions "";
        lualine_x = mkComponentOptions "";
        lualine_y = mkComponentOptions "";
        lualine_z = mkComponentOptions "";
      };
```

File: plugins/statuslines/lualine.nix [hunk 2]
```diff
@@ -173,20 +188,21 @@ in {
         })
         "Sections configuration";
 
-      tabline =
+      inactive_sections =
```
Comment (GaetanLepage) [inline plugins/statuslines/lualine.nix:191]:
```suggestion
      inactiveSections =
```

File: plugins/statuslines/lualine.nix [hunk 3]
```diff
@@ -173,20 +188,21 @@ in {
...
         helpers.mkNullOrOption
```
Comment (GaetanLepage) [inline plugins/statuslines/lualine.nix:192]:
```suggestion
        helpers.mkCompositeOption "Inactive Sections configuration" {
          lualine_c = mkComponentOptions "filename";
          lualine_x = mkComponentOptions "location";
        };
```

File: plugins/statuslines/lualine.nix [hunk 4]
```diff
@@ -173,20 +188,21 @@ in {
...
           };
         })
-        "Tabline configuration";
+        "Inactive Sections configuration";
+
+      tabline = mkEmptySectionOption "Tabline configuration";
+
+      winbar = mkEmptySectionOption "Winbar configuration";
+
+      inactive_winbar = mkEmptySectionOption "Inactive Winbar configuration";
```
Comment (GaetanLepage) [inline plugins/statuslines/lualine.nix:205]:
```suggestion
      inactiveWinbar = mkEmptySectionOption "Inactive Winbar configuration";
```

File: plugins/statuslines/lualine.nix [hunk 5]
```diff
@@ -233,7 +249,10 @@ in {
       };
 
       sections = mapNullable processSections cfg.sections;
+      inactive_sections = mapNullable processSections cfg.inactive_sections;
```
Comment (GaetanLepage) [inline plugins/statuslines/lualine.nix:252]:
```suggestion
      inactive_sections = mapNullable processSections cfg.inactiveSections;
```

File: plugins/statuslines/lualine.nix [hunk 6]
```diff
@@ -233,7 +249,10 @@ in {
...
       tabline = mapNullable processSections cfg.tabline;
+      winbar = mapNullable processSections cfg.winbar;
+      inactive_winbar = mapNullable processSections cfg.inactive_winbar;
```
Comment (GaetanLepage) [inline plugins/statuslines/lualine.nix:255]:
```suggestion
      inactive_winbar = mapNullable processSections cfg.inactiveWinbar;
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Hello, 
Here are a few minor requested changes.

**Comment by GaetanLepage** [comment]
Thanks for the corrections.
One last thing: could you add a test please ?

1. create `tests/test-sources/plugins/statuslines/lualine.nix`
2. Create two (or three) configurations:
  - An `empty` one, where you just enable the plugin
  - A `defaults` one, where you set all the options to the plugin's defaults.
  (- An `example` one where you provide a sensible config)

You can have a look to other test files to see how it works.

**Comment by andrevmatos** [comment]
@GaetanLepage done; I'm not too much into Nix testing, but I hope this should be enough. Thanks

**Comment by GaetanLepage** [comment]
> @GaetanLepage done; I'm not too much into Nix testing, but I hope this should be enough. Thanks

Perfect, thanks !

**Review by GaetanLepage — APPROVED** [review]
Everything looks good now

**Pull request merged**